### PR TITLE
not expose object detail when creating TokenRequest

### PR DESCRIPTION
**Pull request by CaoShuFeng**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/assign @liggitt @mikedanese 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/registry/core/serviceaccount/storage/token.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func (r *TokenREST) Create(ctx genericapirequest.Context, name string, obj runti
 			return nil, errors.NewBadRequest(fmt.Sprintf("cannot bind token to object of type %s", gvk.String()))
 		}
 		if ref.UID != "" && uid != ref.UID {
-			return nil, errors.NewConflict(schema.GroupResource{Group: gvk.Group, Resource: gvk.Kind}, ref.Name, fmt.Errorf("the UID in the bound object reference (%s) does not match the UID in record (%s). The object might have been deleted and then recreated", ref.UID, uid))
```
Comment (CaoShuFeng) [inline pkg/registry/core/serviceaccount/storage/token.go]: We are not sure the user are authoried to read pod.Spec.ServiceAccountName pod.UID or secert.UID.
So, we should not let users known this information, I think.

**Comment by CaoShuFeng** [comment]
kindly ping @liggitt 
Is this patch right?

File: pkg/registry/core/serviceaccount/storage/token.go (hunk 1 above)
Comment (liggitt) [reply pkg/registry/core/serviceaccount/storage/token.go]: I think this should still return a conflict, even if it doesn't echo back the current UID

Comment (CaoShuFeng) [reply pkg/registry/core/serviceaccount/storage/token.go]: Done.

**Pull request merged**